File: src/yh_divisor/scripts/yh_divisor_pub.py
#!/usr/bin/python
# -*- coding: utf-8 -*-

import rospy
from yh_divisor.msg import yh_divisor_msg

def talker():
    pub = rospy.Publisher("yh_divisor_topic",yh_divisor_msg,queue_size= 100)
    rospy.init_node("yh_divisor_pub",anonymous= True)
    LoopRate = rospy.Rate(2)
    cnt = 0
    msg = yh_divisor_msg()
                                                                                                                                                                                                            
    while not rospy.is_shutdown():
        msg.stamp = rospy.get_rostime()
        msg.data = cnt
        
        rospy.loginfo("send message: %d", msg.stamp.secs)
        rospy.loginfo("send message: %d", msg.stamp.nsecs)
        rospy.loginfo("send message: %d", msg.data)
        cnt +=1
        
        pub.publish(msg)
        LoopRate.sleep()
        
if __name__ == "__main__":
    
    try:
        talker()
    except rospy.ROSInternalException:
        pass
        

File: src/yh_divisor/src/yh_divisor_pub.cpp
#include "ros/ros.h" //ros
#include "yh_divisor/yh_divisor_msg.h" //msg After build Auto Create



int main(int argc, char **argv)
{
    ros::init(argc,argv, "yh_divisor_pub"); // node name init
    ros::NodeHandle nh; // node handle
    
    //publisher
    //패키지 topic_tutorial 의 메시지 파일 my_msg 을 이용한
    //퍼블리셔 my_pub 를 작성한다. 토픽은 my_topic 이며,
    //퍼블리셔 큐 사이즈를 100개로 설정
    ros::Publisher my_pub = nh.advertise<yh_divisor::yh_divisor_msg>("yh_divisor_topic", 100);

    ros::Rate loop_rate(1); // loop 10 hz

    yh_divisor::yh_divisor_msg msg;
    int cnt = 0;

    while (ros::ok())
    {
        msg.stamp = ros::Time::now(); // time
        msg.data = cnt;
        //ROS_INFO("Send msg = %d", msg.stamp.sec);
        //ROS_INFO("Send msg = %d", msg.stamp.nsec);
        //ROS_INFO("Send msg = %d", msg.data);
        cnt++;
        my_pub.publish(msg); // msg publish
        loop_rate.sleep();

    }
    

    return 0;
}